uberfire-extensions/uberfire-experimental-extensions/uberfire-experimental-extensions-api/src/main/java/org/uberfire/ext/experimental/service/def/impl/ExperimentalFeatureDefRegistryImpl.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.uberfire.ext.experimental.service.def.impl;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.uberfire.ext.experimental.service.def.ExperimentalFeatureDefinition;
+import org.uberfire.ext.experimental.service.def.ExperimentalFeatureDefRegistry;
+
+public class ExperimentalFeatureDefRegistryImpl implements ExperimentalFeatureDefRegistry {
+
+    protected List<ExperimentalFeatureDefinition> features = new ArrayList<>();
```

Comment:
Instead of using a list and then looking up by traversing all the values in a list, wouldn't it be better to use `Map<String, ExperimentalFeatureDefinition>`? The code would be a bit more conscise and we could avoid potentially problematic states with one experimental feature being in the list more times..